presto-orc/src/main/java/com/facebook/presto/orc/cache/StorageOrcFileTailSource.java

```diff
@@ -152,4 +185,21 @@ private static void checkOrcVersion(OrcDataSource orcDataSource, List<Integer> v
             }
         }
     }
+
+    /**
+     * ORC metadata and DWRF stripe cache sizes are mutually exclusive because
+     * only ORC files have metadata, and only DWRF files have stripe cache.
+     * <p>
+     * Let's check that either both sizes are 0, or only one of them is
+     * greater than 0.
+     */
+    private static void checkSizes(OrcDataSource orcDataSource, int metadataSize, int dwrfStripeCacheReadSize)
+    {
+        if (metadataSize > 0 && dwrfStripeCacheReadSize > 0) {
```

Comment:
also validate dwrfStripeCacheReadSize >= 0. The reason to check here is we want to only fail when dwrfStripeCacheMode is enabled. so as not to make the current existing code fail.

Reply:
I think it would make it too strict. Technically `dwrfStripeCacheSize = 0` is a valid scenario even if the `mode is != NONE`- in this case we won't use the cache and fall back to the reading indexes and footers from the stripes.

Reply:
Added an independent check to verify `dwrfStripeCacheReadSize >= 0` regardless of metadataSize.